SubmittingPatches.rst

```diff
@@ -62,468 +66,207 @@ then you just add a line saying ::
+the length of PR titles, but be aware that they are subject to editing as part
+of the release process.)
 
-The ``git send-email`` command make it super easy to send patches
-(particularly those prepared with git format patch). It is careful to
-format the emails correctly so that you don't have to worry about your
-email client mangling whitespace or otherwise screwing things up. It
-works like so:
+If you believe your changes should be backported to stable branches after the PR
+is merged, open a tracker issue at https://tracker.ceph.com explaining:
 
-.. code-block:: bash
+1. what bug is fixed
+2. why does the bug need to be fixed in <release>
 
-   $ git send-email --to ceph-devel@vger.kernel.org my.patch
+and fill out the Backport field in the tracker issue. For example::
 
-for a single patch, or
+    Backport: mimic, nautilus
 
-.. code-block:: bash
+For information on how backports are done in the Ceph project, refer to the
+document "SubmittingChanges-backports.rst".
 
-   $ git send-email --to ceph-devel@vger.kernel.org mything
 
-to send a whole patch series (prepared with, say, git format-patch).
+.. _`fixes line`:
 
+Fixes line(s)
+-------------
 
-3. Describe your changes
-------------------------
+If the commit fixes one or more issues tracked by http://tracker.ceph.com,
+add a ``Fixes:`` line (or lines) to the commit message, to connect this change
+to addressed issue(s) - for example::
 
-Describe the technical detail of the change(s) your patch includes.
+     Fixes: http://tracker.ceph.com/issues/12345
 
-.. _title_of_commit:
+This line should be added just before the ``Signed-off-by:`` line (see `Sign
+your work`_).
 
-Title of pull requests and title of commits
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+It helps the reviewer to get more context of this bug, so she/he can hence
+update the issue on the bug tracker accordingly.
 
-The text up to the first empty line in a commit message is the commit
-title. Ideally it is a single short line of at most 72 characters,
-summarizing the change. It is required to prefix it with the
-subsystem or module you are changing. For instance, the prefix
-could be "doc:", "osd:", or "common:". One can use::
+So a typical commit message for revising the document could look like::
 
-     git log
+     doc: add "--foo" option to bar
 
-for more examples. It is also conventional to use the imperative mood in the
-commit title. For example::
+     * update the man page for bar with the newly added "--foo" option.
+     * fix a typo
 
-     mds: add perf counter for finisher of MDSRank
+     Fixes: http://tracker.ceph.com/issues/12345
+     Signed-off-by: Random J Developer <random@developer.example.org>
 
-or::
+If a commit fixes a regression introduced by a different commit, please
+add a line referencing the SHA1 of the commit that introduced the regression.
+For example::
 
-     osd: make the ClassHandler::mutex private
+     Fixes: 9dbe7a003989f8bb45fe14aaa587e9d60a392727
```

Comment:
I don't we have ever practised this, at least not in the main repo.  Not saying that this is a bad idea, just that it doesn't codify existing practice and should be discussed on the mailing list first for it to get any traction.  Because everyone is doing their own thing: I noticed @batrick has been using Introduced-by, I usually put an abbreviated SHA1 along with the title somewhere in the commit message, etc.

Reply:
We recently started making a point of mentioning the commit that introduced the problem to help identify follow-on fixes to backport, but I don't think we standardized on the Prefix: that precedes it.  Fixes: seems appropriate though?  In any case, this is a good time to go ahead codify this!

Reply:
@idryomov This format was suggested by Sage quite some months (or even a year or more) ago.

Reply:
I'm down with this format.

Reply:
+1